### PR TITLE
Support ConfigMap mounts in Helm Chart

### PR DESCRIPTION
**Pull request by deepak-shivamurthy**
Support adding ConfigMap mounts to Alluxio pods, like krb5.conf or hdfs config.

### CODE REVIEW

**Review by nirav-chotai — COMMENTED** [review]
Adding `configMaps` should be very similar to `secrets` and allow user to mount one of more `configMaps` to `master`, `worker` and `logserver`

**Comment by nirav-chotai** [comment]
> The krb5.conf file contains Kerberos configuration information required when master/worker nodes are running in order to validate credentials. Hence this should be mounted as ConfigMap.

`krb5.conf` is just an example, in general, there should be a way to create one or mounts with ConfigMaps, very similar to Secrets

**Comment by nirav-chotai** [comment]
@madanadit , @jiacheliu3 , @yuzhu , can you guys approve workflow?

**Comment by yuzhu** [comment]
@ZhuTopher could you take a look pls?

**Review by ZhuTopher — CHANGES_REQUESTED** [review]
Thanks for waiting, I've been out-of-action for a while so I appreciate your patience with getting a reviewer.

I agree with the proposal to allow users to mount `ConfigMap`-volumes like we do with secrets. It'd be nice to make it configurable whether or not they're mounted as `readOnly` or not. Otherwise LGTM

File: integration/kubernetes/helm-chart/alluxio/values.yaml [hunk 1]
```diff
@@ -481,6 +481,19 @@ fuse:
...
+##  ConfigMaps ##
+
+# Format: (<name>:<mount path under /configmaps/>):
+# configmaps:
+#   master: # Shared by master and jobMaster containers
+#     alluxio-krb5-config: krb5Config
+#   worker: # Shared by worker and jobWorker containers
+#     alluxio-krb5-config: krb5Config
+#   logserver: # Used by the logserver container
+#     alluxio-krb5-config: krb5Config
```
Comment (jiacheliu3) [inline integration/kubernetes/helm-chart/alluxio/values.yaml:494]:
```suggestion
#   master: # Shared by master and jobMaster containers
#     alluxio-hdfs-config: hdfsConfig
#   worker: # Shared by worker and jobWorker containers
#     alluxio-hdfs-config: hdfsConfig
#   logserver: # Used by the logserver container
#     alluxio-hdfs-config: hdfsConfig
```

**Review by jiacheliu3 — APPROVED** [review]
LGTM with a minor comment. Thanks for the improvements! Also PTAL at comments from @ZhuTopher

**Comment by jiacheliu3** [comment]
hey @deepak-shivamurthy just wanted to double check if you have signed the CLA. We require contributor to sign that as an agreement for making contributions to our open source code. Thx :)

**Comment by nirav-chotai** [comment]
> hey @deepak-shivamurthy just wanted to double check if you have signed the CLA. We require contributor to sign that as an agreement for making contributions to our open source code. Thx :)

Can you send him the link to sign it? his email is [shivamurthy.deepak@gmail.com](mailto:shivamurthy.deepak@gmail.com)

**Comment by jiacheliu3** [comment]
hey @Xenorith do you know how we sign the CLA after the pipeline got broken?

**Comment by Xenorith** [comment]
yes, we just got this somewhat up and running again. please download the CLA pdf here: https://www.alluxio.io/app/uploads/2022/07/Contribution-License-Agreement-2022-Template.pdf

and send the signed copy to cla@alluxio.org

@Zijian-Zhu @HelloHorizon fyi

**Comment by jiacheliu3** [comment]
> 

Without a digital signature everything becomes so hard, would it be possible to bring the pipeline back then we merge this PR?

File: integration/kubernetes/helm-chart/alluxio/values.yaml (hunk 1 above)
Comment (deepak-shivamurthy) [reply integration/kubernetes/helm-chart/alluxio/values.yaml]: made changes as per the above comment @jiacheliu3

**Comment by deepak-shivamurthy** [comment]
> Thanks for waiting, I've been out-of-action for a while so I appreciate your patience with getting a reviewer.
> 
> I agree with the proposal to allow users to mount `ConfigMap`-volumes like we do with secrets. It'd be nice to make it configurable whether or not they're mounted as `readOnly` or not. Also please make sure to pull the latest version of the repo and update `CHANGELOG.md` and `Chart.yaml` with the bumped chart version number. Otherwise LGTM.

@ZhuTopher updated `CHANGELOG.md` and `Chart.yaml`

**Comment by Xenorith** [comment]
> > 
> 
> @Xenorith sent email with signed CLA, got below response the group you tried to contact (cla) may not exist, or you may not have permission to post messages to the group

thanks for the details; i've fixed the visibility of the group. can you resend?

**Comment by deepak-shivamurthy** [comment]
> cla@alluxio.org

done, thanks @Xenorith

**Comment by deepak-shivamurthy** [comment]
> > > 
> > 
> > 
> > @Xenorith sent email with signed CLA, got below response the group you tried to contact (cla) may not exist, or you may not have permission to post messages to the group
> 
> thanks for the details; i've fixed the visibility of the group. can you resend?

done, thanks @Xenorith

**Review by jiacheliu3 — APPROVED** [review]
LGTM, thanks for the improvements!

**Comment by jiacheliu3** [comment]
@ZhuTopher PTAL thanks :)

**Review by nirav-chotai — APPROVED** [review]

**Comment by nirav-chotai** [comment]
@ZhuTopher / @jiacheliu3 , can this be merged?

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Pull request merged**